**Security review** (on behalf of @jdomeracki-coder, generated by Coder Agents)

No blocking security issues.

**Positive changes:**

- Replacing the unquoted `$ALL_CHANGED_FILES` expansion (old `# shellcheck disable=SC2086`) with the piped `echo | tr | xargs -d '\n'` pattern eliminates word-splitting and glob-expansion on filenames. Clear improvement.
- Step outputs are consumed through `env:` context rather than `${{ }}` template injection in `run:`, which is the correct pattern to prevent expression injection from attacker-controlled filenames.

**Nit (non-blocking):**

- `echo "$ALL_CHANGED_FILES"` will misbehave if a value ever starts with `-e` or `-n` (bash's builtin `echo` interprets these as flags). `printf '%s\n' "$ALL_CHANGED_FILES"` is strictly safer. Negligible risk for `.md` paths, but worth noting as a hardening opportunity.

--- a/.github/workflows/docs-ci.yaml
+++ b/.github/workflows/docs-ci.yaml
@@ -48,17 +48,20 @@ jobs:
       # comma-joined string as a single non-matching glob, and
       # `markdown-table-formatter` would fall back to scanning every `.md` in
       # the working tree when invoked with no positional args.
+      #
+      # `printf '%s\n'` is used instead of `echo` so a hypothetical leading
+      # `-e` or `-n` in a path is treated as data, not a bash builtin flag.
 
       - name: lint
         if: steps.changed-md.outputs.any_changed == 'true'
         run: |
-          echo "$ALL_CHANGED_FILES" | tr ',' '\n' | xargs -d '\n' pnpm exec markdownlint-cli2
+          printf '%s\n' "$ALL_CHANGED_FILES" | tr ',' '\n' | xargs -d '\n' pnpm exec markdownlint-cli2
         env:
           ALL_CHANGED_FILES: ${{ steps.changed-md.outputs.all_changed_files }}
 
       - name: fmt
         if: steps.changed-md.outputs.any_changed == 'true'
         run: |
-          echo "$ALL_CHANGED_FILES" | tr ',' '\n' | xargs -d '\n' pnpm exec markdown-table-formatter --check
+          printf '%s\n' "$ALL_CHANGED_FILES" | tr ',' '\n' | xargs -d '\n' pnpm exec markdown-table-formatter --check
         env:
           ALL_CHANGED_FILES: ${{ steps.changed-md.outputs.all_changed_files }}

